pkg/controller/status/status.go

```diff
@@ -670,8 +674,44 @@ func (m *statusManager) degradedReason() string {
 	return strings.Join(reasons, "; ")
 }
 
-func hasPendingCSR(ctx context.Context, cli client.Client, labelMap map[string]string) (bool, error) {
-	csrs := &v1beta1.CertificateSigningRequestList{}
+func hasPendingCSR(ctx context.Context, m *statusManager, labelMap map[string]string) (bool, error) {
+	if m.kubernetesVersion.Major > 1 || (m.kubernetesVersion.Major == 1 && m.kubernetesVersion.Minor >= 19) {
+		return hasPendingCSRUsingCertV1(ctx, m.client, labelMap)
+	}
+	return hasPendingCSRUsingCertV1beta1(ctx, m.client, labelMap)
+}
+
+func hasPendingCSRUsingCertV1(ctx context.Context, cli client.Client, labelMap map[string]string) (bool, error) {
+	csrs := &certV1.CertificateSigningRequestList{}
+	selector := labels.SelectorFromSet(labelMap)
+	err := cli.List(ctx, csrs, &client.ListOptions{LabelSelector: selector})
+	if err != nil {
```

Comment:
From this point onwards the code is duplicate from the other func if you consider that constant `certV1beta1.CertificateApproved` == `certV1.CertificateApproved`. I think you can turn the two funcs into one  if you add a bool param for `useV1Api` or something like that.

Reply:
But this `csr` is package-specific, so are `csr.Status`, `csr.Status.Conditions` and `condition.Type`. Unless we define a common interface for both `v1.CertificateSigningRequest` and `v1beta1.CertificateSigningRequest`, combining them into one function will make the code very ugly.